src/Host/Broker/Impl/Startup/Program.cs

```diff
@@ -0,0 +1,114 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.IO;
+using System.IO.Pipes;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Hosting.Server.Features;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Microsoft.R.Host.Broker.Logging;
+using Newtonsoft.Json;
+
+namespace Microsoft.R.Host.Broker.Startup {
+    public class Program {
+        private static readonly ILoggerFactory _loggerFactory = new LoggerFactory();
+        private static ILogger _logger;
+        private static readonly StartupOptions _startupOptions = new StartupOptions();
+        private static readonly CancellationTokenSource _cts = new CancellationTokenSource();
+
+        internal static IConfigurationRoot Configuration { get; private set; }
+
+        public static CancellationToken CancellationToken => _cts.Token;
+
+        static Program() {
+        }
+
+        public static void Main(string[] args) {
+            var configBuilder = new ConfigurationBuilder().AddCommandLine(args);
+            Configuration = configBuilder.Build();
+
+            string configFile = Configuration["config"];
+            if (configFile != null) {
+                configBuilder.AddJsonFile(configFile, optional: false);
+                Configuration = configBuilder.Build();
+            }
+
+            ConfigurationBinder.Bind(Configuration.GetSection("startup"), _startupOptions);
+
+            _loggerFactory
+                .AddDebug()
+                .AddConsole(LogLevel.Trace)
+                .AddProvider(new FileLoggerProvider(_startupOptions.Name));
+            _logger = _loggerFactory.CreateLogger<Program>();
+
+            if (_startupOptions.Name != null) {
+                _logger.LogInformation($"Broker name '{_startupOptions.Name}' assigned");
+            }
+
+            CreateWebHost().Run();
+        }
+
+        public static IWebHost CreateWebHost() {
+            var webHostBuilder = new WebHostBuilder()
+                .UseLoggerFactory(_loggerFactory)
+                .UseConfiguration(Configuration)
+                .UseKestrel(options => {
+                    //options.UseConnectionLogging();
```

Comment:
This actually logs entire HTTP traffic (including contents), so it's very verbose, and should not be on by default; but can be helpful occasionally when debugging, so keeping it commented out for that purpose.